Sources/StreamChat/Utils/Codable+Extensions.swift

```diff
@@ -17,12 +18,21 @@ final class StreamJSONDecoder: JSONDecoder, @unchecked Sendable {
         let dateCache = NSCache<NSString, NSDate>()
         dateCache.countLimit = 5000 // We cache at most 5000 dates, which gives good enough performance
 
-        self.init(dateFormatter: iso8601formatter, dateCache: dateCache)
+        self.init(
+            dateFormatter: iso8601formatter,
+            dateCache: dateCache,
+            rawJSONCache: RawJSONCache(countLimit: 500)
```

Comment:
In the performance benchmark table, we are not checking the memory usage. Is there any significant increase in memory by caching all the extra data?

Reply:
I don't expect it, but maybe we clean it up on memory warning notifications, just in case?

Reply:
Internally it is NSCache which clears itself on low memory warning. I used a wrapping class because of unit-tests. Can't mock NSCache from the Swift side (it is objc generic class and brings restrictions).

Reply:
There is significant decrease in memory usage used for RawJSON.
Before:
<img width="1129" alt="NoCache" src="https://github.com/user-attachments/assets/de14af65-f7ca-49c0-9bba-8f8614b88d1c" />
After
<img width="1052" alt="Cache" src="https://github.com/user-attachments/assets/39a7743f-5541-46b6-9da7-bed3bb054642" />